llvm/lib/Target/RISCV/GISel/RISCVLegalizerInfo.cpp

```diff
@@ -521,19 +523,22 @@ RISCVLegalizerInfo::RISCVLegalizerInfo(const RISCVSubtarget &ST)
       .legalFor(ST.hasStdExtD(), {{s32, s64}})
       .legalFor(ST.hasStdExtZfh(), {{s16, s32}})
       .legalFor(ST.hasStdExtZfh() && ST.hasStdExtD(), {{s16, s64}})
-      .libcallFor({{s32, s64}});
+      .libcallFor({{s32, s64}})
+      .libcallFor(ST.is64Bit(), {{s32, s128}, {s64, s128}});
```

Comment:
Why aren't these ok on 32-bit? They return 32/64-bit values, rather than 128-bit, right?

Maybe you're waiting so the FPTRUNC/FPEXT definitions can be symmetric? If so, that's reasonable.

Reply:
I didn't think too hard about it and I didn't make a rv32 test. I can split off the rv32 subset into a new test and fix this.